ipaserver/plugins/idviews.py

```diff
@@ -91,6 +91,27 @@
 )
 
 
+def normalize_idview_name(value):
+    if value in (None, '',):
+        return DEFAULT_TRUST_VIEW_NAME
+    return value
+
+
+def handle_idoverride_memberof(self, ldap, dn, found, not_found,
+                               *keys, **options):
+    assert isinstance(dn, DN)
+    if 'idoverrideuser' in options:
+        for id in found['member']['idoverrideuser']:
+            try:
+                e = ldap.get_entry(id, ['*'])
+            except errors.NotFound:
+                # We are not adding an object here, only modifying existing
+                continue
+            e = add_missing_object_class(ldap, 'nsmemberof',
+                                         id, entry_attrs=e, update=True)
```

Comment:
I'd add a call to has_objectclass() to avoid fetching the entry twice only to find out it already has nsmemberof.

You don't need to save the value of the entry in either case, e is unused.

I guess this potentially could result in adding an object class to an entry and then fail later?

Reply:
Yeah, not sure how I missed that. You will save another fetch if you call has_objectclass() first because the first thing that add_missing_object_class does is fetch the entry again.

Reply:
Yes, I have it in my pending updates. Need to complete tests and design document update first.